src/components/TodoFilter/TodoFilter.tsx

```diff
@@ -1,30 +1,72 @@
-export const TodoFilter = () => (
-  <form className="field has-addons">
-    <p className="control">
-      <span className="select">
-        <select data-cy="statusSelect">
-          <option value="all">All</option>
-          <option value="active">Active</option>
-          <option value="completed">Completed</option>
-        </select>
-      </span>
-    </p>
+import React from 'react';
+import { Filters } from '../../types/Filters';
 
-    <p className="control is-expanded has-icons-left has-icons-right">
-      <input
-        data-cy="searchInput"
-        type="text"
-        className="input"
-        placeholder="Search..."
-      />
-      <span className="icon is-left">
-        <i className="fas fa-magnifying-glass" />
-      </span>
+const filters = ['All', 'Active', 'Completed'];
 
-      <span className="icon is-right" style={{ pointerEvents: 'all' }}>
-        {/* eslint-disable-next-line jsx-a11y/control-has-associated-label */}
-        <button data-cy="clearSearchButton" type="button" className="delete" />
-      </span>
-    </p>
-  </form>
-);
+type Props = {
+  handleFilterChange: React.Dispatch<React.SetStateAction<Filters>>;
+  setQuery: React.Dispatch<React.SetStateAction<string>>;
+  query: string;
+  selectedFilter: Filters;
+};
+
+export const TodoFilter: React.FC<Props> = ({
+  handleFilterChange,
+  setQuery,
+  query,
+  selectedFilter,
+}) => {
+  const onFilterChange = (event: React.ChangeEvent<HTMLSelectElement>) => {
+    handleFilterChange(event.target.value as Filters);
```

Comment:
The `onFilterChange` function casts `event.target.value` directly to `Filters`, which may not align with the `Filters` enum. Consider mapping the string values from the `filters` array to the corresponding `Filters` enum values to ensure correct behavior.